### ISSUE TITLE
Optimize Updates to Lists

### ISSUE THREAD

**Issue by semihsalihoglu-uw** [kuzudb/kuzu#730]
Our update logic to lists is being done in an unoptimized manner at this stage. Our primary goal is not performance but correctness. There are many places where we can improve the performance of updates:

[ ] Implement updating lists in batches.
[ ] Make CSRs with 1.2x or some other ratio gaps. That will allow localizing updates. This requires changes to both the loader and the updates.
[ ] Support updating large lists without re-writing the entire list in Unstructured Properties (at the time of writing this issue updates to AdjLists and PropertyLists are not implemented, so whether this applies to them is not yet clear).
[ ] Parallelize prepareCommitOrRollbackIfNecessary where we implement the CSR-reconstruction algorithm. This could be implemented by creating a simple plan upon connection->commit() (which is what eventually calls prepareCommitOrRollbackIfNecessary) that puts each UnstructuredPropertyList that needs to be updated into the task scheduler, which grab morsels and performs the preparation of updates to the chunks in parallel.
[ ] We cannot make large lists small again. Any list that became large remains large.
[ ] Manage free pages in lists (https://github.com/graphflowdb/graphflowdb/issues/731).

**Comment by ray6080** [comment]
This is no longer relevant.
